platforms/unix/vm/sqUnixSpurMemory.c

```diff
@@ -164,27 +164,27 @@ sqDeallocateMemorySegmentAtOfSize(void *addr, sqInt sz)
 void
 sqMakeMemoryExecutableFromTo(unsigned long startAddr, unsigned long endAddr)
 {
+	unsigned long size = endAddr - startAddr;
 	unsigned long firstPage = roundDownToPage(startAddr);
 	if (mprotect((void *)firstPage,
-				 endAddr - firstPage + 1,
+				 size,
 				 PROT_READ | PROT_WRITE | PROT_EXEC) < 0)
 		perror("mprotect(x,y,PROT_READ | PROT_WRITE | PROT_EXEC)");
 }
 
 void
 sqMakeMemoryNotExecutableFromTo(unsigned long startAddr, unsigned long endAddr)
 {
-#	if 0
+	unsigned long size = endAddr - startAddr;
 	unsigned long firstPage = roundDownToPage(startAddr);
 	/* Arguably this is pointless since allocated memory always includes write
 	 * permission by default.  Annoyingly the mprotect call fails on both linux
 	 * and mac os x.  So make the whole thing a nop.
 	 */
 	if (mprotect((void *)firstPage,
-				 endAddr - firstPage + 1,
```

Comment:
man mprotect https://linux.die.net/man/2/mprotect : it requires an address aligned on page boundary
There is no such requirement for VirtualProtect https://msdn.microsoft.com/en-us/library/windows/desktop/aa366898(v=vs.85).aspx

I see from msdn page that we should FlushInstructionCache before executing any newly installed code and I fail to see any such call (grep -r FlushInstructionCache platforms). Unless the flush is performed by another mean?